src/emailaction.cpp

```diff
@@ -715,6 +715,43 @@ QMailAccountId Synchronize::accountId() const
 }
 
 /*
+    TransmitMessage
+*/
+TransmitMessage::TransmitMessage(QMailTransmitAction* transmitAction, const QMailMessageId &messageId)
+    : EmailAction()
+    , _transmitAction(transmitAction)
+    , _messageId(messageId)
+{
+    _description = QString("transmit-message:message-id=%1").arg(_messageId.toULongLong());
+    _type = EmailAction::Transmit;
+}
+
+TransmitMessage::~TransmitMessage()
+{
+}
+
+void TransmitMessage::execute()
+{
+    _transmitAction->transmitMessage(_messageId);
+}
+
+QMailServiceAction* TransmitMessage::serviceAction() const
+{
+    return _transmitAction;
+}
+
+QMailMessageId TransmitMessage::messageId() const
+{
+    return _messageId;
+}
+
+QMailAccountId TransmitMessage::accountId() const
+{
+    QMailMessage msg(_messageId);
```

Comment:
Yeah makes sense since that object is substantially simpler and smaller, will create task to check in order places, had plans to check it but then forgot :)